impl/src/main/java/org/jboss/weld/bootstrap/Validator.java

```diff
@@ -922,16 +903,23 @@ private static void reallyValidatePseudoScopedBean(Bean<?> bean, BeanManagerImpl
 
     private static void validatePseudoScopedInjectionPoint(InjectionPoint ij, BeanManagerImpl beanManager, Set<Object> dependencyPath, Set<Bean<?>> validatedBeans) {
         Set<Bean<?>> resolved = beanManager.getBeans(ij);
-        Bean<?> bean = beanManager.resolve(resolved);
-        if (bean != null) {
-            if (!(bean instanceof AbstractBuiltInBean<?>)) {
-                if (!ij.isDelegate()) {
-                    boolean normalScoped = beanManager.isNormalScope(bean.getScope());
-                    if (!normalScoped && !(bean instanceof SessionBean)) {
-                        reallyValidatePseudoScopedBean(bean, beanManager, dependencyPath, validatedBeans);
+        try {
+            Bean<?> bean = beanManager.resolve(resolved);
+            if (bean != null) {
+                if (!(bean instanceof AbstractBuiltInBean<?>)) {
+                    if (!ij.isDelegate()) {
+                        boolean normalScoped = beanManager.isNormalScope(bean.getScope());
+                        if (!normalScoped && !(bean instanceof SessionBean)) {
+                            reallyValidatePseudoScopedBean(bean, beanManager, dependencyPath, validatedBeans);
+                        }
                     }
                 }
             }
+        } catch (AmbiguousResolutionException ex) {
```

Comment:
I would only use the `try-catch` block for `beanManager.resolve()` to make it clearer.